identity-wallet/src/state/credentials/reducers/vp_token_payload_prep.rs

```diff
@@ -0,0 +1,109 @@
+use crate::error::AppError;
+use chrono::{Duration, Utc};
+use identity_core::common::Object as IotaObject;
+
+use identity_credential::{credential::Jwt, presentation::Presentation};
+use identity_iota::core::Url;
+use identity_iota::did::{CoreDID, DIDUrl, DID};
+use jsonwebtoken::Algorithm;
+use jsonwebtoken::Header;
+use oid4vc::oid4vc_core::authorization_request::{AuthorizationRequest, Object};
+use oid4vc::oid4vc_core::{jwt, Subject};
+use oid4vc::oid4vp::dcql::dcql_query::{CredentialQuery, Format};
+use oid4vc::oid4vp::oid4vp::OID4VP;
+use oid4vc::oid4vp::token::{
+    verifiable_presentation_jwt::VerifiablePresentationJwt,
+    vp_token::{PresentationFormat, VpToken},
+    vp_token_builder::VpTokenBuilder,
+};
+use serde_json::Value;
+use std::sync::Arc;
+
+pub async fn prepare_vp_token_object(
+    selected_verifiable_credentials: Vec<(CredentialQuery, Value)>,
+    subject_did: &CoreDID,
```

Comment:
Instead of expecting `subject_did: &CoreDID` here, I think it would be better to expect `did_method: &str` because you need `did_method` for deriving the key_id (see [this](https://github.com/impierce/identity-wallet/pull/578/files#r2288364199)) and you can use it to derive `subject_did`.

This way then you can also get rid of [this code here](https://github.com/impierce/identity-wallet/pull/578/files#diff-760e874e5dd88ae1a137211245fd6c0898c533a8eb00b5df7be71f4a8c057c3bR159-R165).